js/classic/ui.js

```diff
@@ -9,6 +9,223 @@
   var utils = exports.classic.utils;
   var ui = exports.classic.ui;
 
+  /**
+   * An UI component represents a switch.
+   *
+   * It supports turing on/off operations.
+   *
+   * @param handler an event handler object, which should include below methods:
+   *     .onTurnOn -- called when the user tries to turn on the switch.
+   *     .onTurnOff -- called when the user tries to turn off the switch.
+   *     .onCancelTurnOn -- called when the user tries to turn off the switch
+   *         while the current state is "turning on".
+   *     .onCancelTurnOff -- called when the user tries to turn on the switch
+   *         while the current state is "turning off".
+   * @param defaultState the default state of this switch.
+   * @param enable whether this switch is enabled or not in default.
+   */
+  ui.SwitchButton = function(handler, defaultState, enable) {
+    this._selector = $(window.document.createElement('div')).slider({max: 2});
+    this._selector.addClass('classic-ui-switch-button');
+    this._selector.slider('disable');
+    this._handler = handler;
+    this._state = '';
+    this._enable = null;
+
+    this._selector.click(this._onClick.bind(this));
+
+    this.state = (defaultState !== undefined ? defaultState : this.State.OFF);
+    this.enable = (enable !== undefined ? enable : true);
+  };
+
+  /**
+   * Enumerate of the states of `ui.SwitchButton`
+   */
+  ui.SwitchButton.State = {
+    OFF: 'off',
+    TURNING_ON: 'turning-on',
+    TURNING_OFF: 'turning-off',
+    ON: 'on'
+  };
```

Comment:
It seems like some kind of constant value. We could utilize `Object.freeze({})` to make sure no one can alter it. That is to say:
```js
ui.SwitchButton.State = Object.freeze({
  OFF: 'off',
  TURNING_ON: 'turning-on',
  TURNING_OFF: 'turning-off',
  ON: 'on'
});
```